src/cswinrt/strings/WinRT.cs

```diff
@@ -49,18 +56,50 @@ internal static unsafe int CoCreateInstance(ref Guid clsid, IntPtr outer, uint c
         [DllImport("api-ms-win-core-com-l1-1-0.dll")]
         internal static extern unsafe int CoIncrementMTAUsage(IntPtr* cookie);
 
-        [DllImport("kernel32.dll", SetLastError = true)]
-        [return: MarshalAs(UnmanagedType.Bool)]
-        internal static extern bool FreeLibrary(IntPtr moduleHandle);
+        internal static bool FreeLibrary(IntPtr moduleHandle)
+        {
+            int lastError;
+            bool returnValue;
+            int nativeReturnValue;
+            {
+                SetLastError(0);
+                nativeReturnValue = PInvoke(moduleHandle);
+                lastError = GetLastError();
+            }
+
+            // Unmarshal - Convert native data to managed data.
+            returnValue = nativeReturnValue != 0;
+            SetLastError(lastError);
```

Comment:
This isn't correct, it's literally just setting the same error it just retrieved from the same place.